htdocs/holiday/class/holiday.class.php

```diff
@@ -898,7 +898,13 @@ public function approve($user = null, $notrigger = 0)
 		if ($checkBalance > 0) {
 			$balance = $this->getCPforUser($this->fk_user, $this->fk_type);
 
-			if ($balance < 0) {
+			$startDdatetime = new DateTime("@$this->date_debut");
+			$endDatetime = new DateTime("@$this->date_fin");
+
+			$interval = $startDdatetime->diff($endDatetime);
+			$days = $interval->days;
+
+			if ($balance - $days < 0) {
```

Comment:
I suggest to add a test on a getDolGlobalString('HOLIDAY_ALLOW_NEGATIVE_BALANCE')  for the case when company need to allow this (a lof of company does not manage the balance but need to follow leave requests.